packages/core/src/fhir-deduplication/comparators.ts

```diff
@@ -0,0 +1,27 @@
+import { Identifier, Resource } from "@medplum/fhirtypes";
+
+export function sameResourceId(resourceA: Resource, resourceB: Resource): boolean {
+  const idOfResourceA = resourceA.id;
+  const idOfResourceB = resourceB.id;
+  if (idOfResourceA === undefined || idOfResourceB === undefined) return false;
+  return idOfResourceA === idOfResourceB;
+}
+
+export function sameResourceIdentifier(
+  resourceA: Resource & { identifier?: Identifier[] },
+  resourceB: Resource & { identifier?: Identifier[] }
+): boolean {
+  const identifiersOfResourceA = resourceA.identifier;
+  const identifiersOfResourceB = resourceB.identifier;
+  if (identifiersOfResourceA === undefined || identifiersOfResourceB === undefined) return false;
+  return identifiersOfResourceA.some(identifierOfResourceA => {
+    return identifiersOfResourceB.some(identifierOfResourceB =>
+      sameIdentifier(identifierOfResourceA, identifierOfResourceB)
+    );
+  });
+}
```

Comment:
anytime we do a doubly nested some we can use intersectionWith: 
`intersectionWith(identifiersOfResourceA, identifiersOfResourceB, sameIdentifier)` which basically does true if there is set intersection based on the comparator you specify. I see it come up often enough that I think its worth having it become a team idiom, because the some(some()) needs to be read more closely in case someone sneaks a ! or additional logic inside.